### PR TITLE
feature: Add loading indicator to keyword filter input

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Enhancement**

This PR resolves [CX-1742]

### Description

https://user-images.githubusercontent.com/4691889/127331410-a48b838a-526e-4e32-b596-19add1b2a2f0.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Show loading spinner on artist auction result keyword filter input - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1742]: https://artsyproduct.atlassian.net/browse/CX-1742

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8d490863390b40e0fa9ca0b0e531f93f7efa3c20
</p>

File: src/lib/Components/Artist/ArtistInsights/ArtistInsightsAuctionResults.tsx [hunk 1]
```diff
@@ -48,6 +54,9 @@ const ArtistInsightsAuctionResults: React.FC<Props> = ({ artist, relay, scrollTo
       relay.refetchConnection(
         PAGE_SIZE,
         (error) => {
+          requestAnimationFrame(() => setKeywordFilterRefetching(false))
+          // setTimeout(() => setKeywordFilterRefetching(false), 5000)
```
Comment (olerichter00) [inline src/lib/Components/Artist/ArtistInsights/ArtistInsightsAuctionResults.tsx:58]:
```suggestion
```

File: src/lib/Components/ArtworkFilter/Filters/KeywordFilter.tsx [hunk 2]
```diff
@@ -27,6 +35,7 @@ export const KeywordFilter: React.FC<KeywordFilterProps> = ({ artistId, artistSl
   const inputRef = useRef(null)
 
   const updateKeywordFilter = (text: string) => {
+    // setTypingToFalse()
```
Comment (olerichter00) [inline src/lib/Components/ArtworkFilter/Filters/KeywordFilter.tsx:38]:
```suggestion
```

File: src/lib/Components/ArtworkFilter/Filters/KeywordFilter.tsx [hunk 3]
```diff
@@ -59,9 +69,14 @@ export const KeywordFilter: React.FC<KeywordFilterProps> = ({ artistId, artistSl
 
   return (
     <Input
+      loading={loading}
       icon={<SearchIcon width={18} height={18} />}
       placeholder="Search by artwork title, series, or description"
-      onChangeText={handleChangeText}
+      onChangeText={(e) => {
+        handleTypingStart()
+
```
Comment (olerichter00) [inline src/lib/Components/ArtworkFilter/Filters/KeywordFilter.tsx:76]:
```suggestion
```

File: src/lib/Components/Artist/ArtistInsights/ArtistInsightsAuctionResults.tsx [hunk 4]
```diff
@@ -48,6 +54,8 @@ const ArtistInsightsAuctionResults: React.FC<Props> = ({ artist, relay, scrollTo
       relay.refetchConnection(
         PAGE_SIZE,
         (error) => {
+          requestAnimationFrame(() => setKeywordFilterRefetching(false))
```
Comment (olerichter00) [inline src/lib/Components/Artist/ArtistInsights/ArtistInsightsAuctionResults.tsx:57]: **Note:** This should delay removing the spinner a bit.

File: src/lib/Components/ArtworkFilter/Filters/KeywordFilter.tsx [hunk 5]
```diff
@@ -39,6 +47,7 @@ export const KeywordFilter: React.FC<KeywordFilterProps> = ({ artistId, artistSl
   }
 
   const handleChangeText = useMemo(() => debounce(updateKeywordFilter, DEBOUNCE_DELAY), [appliedFiltersParams])
+  const handleTypingStart = useMemo(() => throttle(() => onTypingStart?.(), DEBOUNCE_DELAY), [onTypingStart])
```
Comment (olerichter00) [inline src/lib/Components/ArtworkFilter/Filters/KeywordFilter.tsx:50]: **Note:** This sets the loading prop for the input to show the spinner. We don't want to delay (debounce) showing the spinner and throttle instead to make the spinner immediately visible.

**Review by pvinis — APPROVED** [review]

**Pull request merged**